pydicom/dataset.py

```diff
@@ -1073,7 +1074,17 @@ def _slice_dataset(self, start, stop, step):
         if stop is None:
             stop = all_tags[-1] + 1
 
-        slice_tags = [tag for tag in all_tags if Tag(start) <= tag < Tag(stop)]
+        # Issue 92: if stop is None then 0xFFFFFFFF + 1 causes overflow in Tag
+        if stop == 0x100000000:
```

Comment:
Couln't you just use 
``` python
slice_tags = [tag for tag in all_tags if Tag(start) <= tag <= Tag(stop-1)]
```
?

Reply:
Done

Reply:
Sorry for nitpicking, but what I meant was: why don't you use this line for all cases? Is there a need for the second case with `<`?

Reply:
Ok, my bad - you are right of course!

Reply:
No problem, I had to check for myself why it wrote it that way